backend/controllers/hashtag.js

```diff
@@ -2,9 +2,17 @@ const Hashtag = require("./../models/Hashtag");
 
 module.exports.createHashtag = async (req, res) => {
   try {
-    const tag = req.body.tag;
-    const hashtag = await Hashtag.create({tag});
-
+    var tag = req.body.tag;
+    if (tag.charAt(0) != "#") {
+      tag = "#" + tag;
+    }
+    const checkExistingHashtag = await Hashtag.findOne({ where: { tag: tag } });
+    if (checkExistingHashtag) {
```

Comment:
The database will perform a unique check for us while creating the tag.
Ensure that we have a unique index in the model and that should be it.
No need to add an extra query for this.